server/check_output_directory.js

```diff
@@ -16,7 +16,12 @@ async function get_error(directory) {
   if (!fs.statSync(directory).isDirectory()) {
     error = "exists, but is not a directory";
   }
-  const { uid, gid } = os.userInfo();
+  try {
+    const { uid, gid } = os.userInfo();
+  } catch (e) {
+    console.error(`os.userInfo does not work in this environment.`);
+    const { uid, gid } = [];
+  }
```

Comment:
this does not work, does it ? uid and gid become local to the scope of the try/catch.

Looking at the code, we could avoid the call to `userInfo` completely in the normal case, since we use the uid/gid only when the writeFileSync fails.

```js
  try {
    fs.writeFileSync(tmpfile, "{}");
    fs.unlinkSync(tmpfile);
  } catch (e) {
    let err_msg = "does not allow file creation and deletion. ";
    try {
        const { uid, gid } = os.userInfo();
        err_msg += "Check the permissions of the directory, and if needed change them so that " +
      `user with UID ${uid} has access to them. This can be achieved by running the command: chown ${uid}:${gid} on the directory`;
      } finally {
             return err_msg;
      }
  }
```